scanner/src/main/kotlin/Scanner.kt

```diff
@@ -79,6 +84,18 @@ class Scanner(
         require(scannerWrappers.isNotEmpty() && scannerWrappers.any { it.value.isNotEmpty() }) {
             "At least one ScannerWrapper must be provided."
         }
+
+        scannerWrappers.values.flatten().distinct().forEach { scannerWrapper ->
+            scannerWrapper.criteria?.run {
```

Comment:
IMO it would be better to just call `scannerWrapper.criteria?.matches(scannerWrapper.details)` here, this would also validate that the name and configuration match.